docs/search.md

```diff
@@ -0,0 +1,139 @@
+# Search
+
+The Realm object has a `searchIndex` property, which in turn as a `search(...)` method. This is the method used to issue queries and get search results (card instance JSON-API representations) from the realm index.
+
+Note that search results are only as good as the most recent index, so if indexing is not working, or some cards are failing to be indexed, you may not get the results you expect. 
+
+## Search Queries
+
+The `SearchIndex#search(...)` method has one required argument, a search query. The simplest query is an empty object (`{}`), which returns all cards in the index.
+
+### Filters
+
+The Query object may have a `filter` property, which controls which cards are included in the result set.
+
+* `type`: allow specification of a particular card type. Also includes descendant cards in this card's adoption chain.
+* `eq`: allows specifying a card with a specific field value
+* `range`: allows specifying a card with a field value that is gt, gte, lt, or lte a specific value
+* `any`: allows an "OR" union of other filters
+* `every`: allows an "AND" union of other filters
+* `not`: allow negating another filter
+
+* `on`: may be used with `eq` or `range` to limit results to a particular card type.
+
+Nested fields may be specified using a dot (`.`) notation.
```

Comment:
might be worth mentioning that our ability to filter containsMany and linksToMany fields is quite limited. Say for instance we have a card `Company`  with a `department` field that is a linksToMany field. There would be no easy way to express a filter that returns all the companies that have more than 5 departments. So the idea is that better support for plural field predicate filtering is desired.

Reply:
👍